lib/dynamo.js

```diff
@@ -3,9 +3,31 @@
 const AWS = require('aws-sdk');
 const zlib = require('zlib');
 const { BatchGet, BatchWrite } = require('@aws/dynamodb-batch-iterator');
+const logger = require('./logger');
+const sts = new AWS.STS();
+const region = process.env.AWS_REGION || 'us-east-1';
 
-// ddb client (exported for testing/mocking)
-exports.client = new AWS.DynamoDB({region: process.env.AWS_REGION || 'us-east-1'});
+// optionally load a client using a role
+let _client = null;
```

Comment:
I wonder if this is a "bad idea".  Since the client will be memoized across multiple lambda executions, and by default the STS credentials expire in 3600 seconds.

Reply:
Was trying to avoid multiple STS calls per lambda execution.  But maybe that's overthinking it.